lib/TabTracker.js

```diff
@@ -45,6 +52,16 @@ TabTracker.prototype = {
     }
     tabs.removeListener("open", this.onOpen);
     Services.obs.removeObserver(this.onPerfLoadComplete, PERF_LOG_COMPLETE_NOTIF);
+    Services.obs.removeObserver(this.onCacheUpdate, CACHE_UPDATE_NOTIF);
```

Comment:
Well,  we UPDATE our values on cache updates.  So there's always something in the tabData, right?
And on the cache update, that something changes to the the current history and bookmark size, which seems reasonably OK.   Now, this indeed relies on the fact that we update cache every time something places change happens, but .....   is this a problem?